cmd/codegen/generator/go/templates/module_funcs.go

```diff
@@ -276,6 +289,26 @@ func (ps *parseState) parseParamSpecVar(field *types.Var, docComment string, lin
 			optional, _ = strconv.ParseBool(v)
 		}
 	}
+	defaultPath := ""
+	if v, ok := pragmas["defaultPath"]; ok {
+		defaultPath = v
+		if strings.HasPrefix(v, `"`) && strings.HasSuffix(v, `"`) {
+			defaultPath = v[1 : len(v)-1]
+		}
+
+		optional = true // If defaultPath is set, the argument becomes optional
```

Comment:
Hm, this doesn't match the behavior of `default`. Definitely, the current behavior of needing `optional` and `default` is absolutely weird, but if we fix this, I think we should fix this consistently.

Reply:
We do plan on fixing this, see:
- https://github.com/dagger/dagger/issues/6749#issuecomment-2078334724

Simple enough to add to this release.

Reply:
Reminder that `optional = true` here, actually means _nullable_, but in this case it's a lazy default, i.e., it doesn't get saved in the schema as the actual default. It makes sense for that to be `null`. It's at runtime that the engine sees that the value is _null_, so it resolves `defaultPath` and feeds that as input.

With a normal `// +default`, it's different. If you always assume that always means it's `optional = true`, you're actually saying any arg with a default value is also nullable.